openwisp_monitoring/device/tests/test_settings.py

```diff
@@ -0,0 +1,31 @@
+from unittest.mock import patch
+
+from django.core.exceptions import ImproperlyConfigured
+
+from . import DeviceMonitoringTestCase
+
+
+# TODO: The tests are validating correctly but coverage
+# remains unaffected by them, figure out why?
```

Comment:
use `ipdb` to inspect the code and brute force it until you find out what is happening

Reply:
> Use the brute-force, young padawan.

Learning from the great Jedi :blush: 

I used `ipdb` to inspect and found out the code was raising the error `ImproperlyConfigured` but `ipdb` did't show that any line of `settings.py` got executed while running the test.

After some thoughts I realized `settings.py` was read as a string and then being executed line by line by **exec()** in my previous commit, so `settings.py` was never imported or executed :sweat_smile:. After that, I tried with other options like `os.system()`, `subprocess()` but neither worked so finally did a bad workaround (I don't think it's a nice way of doing it but lines are getting covered), please check it out